src/main/java/org/eclipse/uprotocol/cloudevent/factory/UCloudEvent.java

```diff
@@ -398,22 +400,38 @@ static UMessage toMessage(CloudEvent event) {
      * @return returns the cloud event
      */
     static CloudEvent fromMessage(UMessage message) {
-        UAttributes attributes = message.getAttributes();
+        return fromMessageParts(message.getSource(), message.getAttributes(), message.getPayload());
+    }
+
+
+    /**
+     * Get the Cloudevent from the UMessage Parts (UUri, UAttributes, and UPayload) <br>
+     * <b>Note: For now, only the value format of UPayload is supported in the SDK.If the UPayload has a reference, it
+     * needs to be copied to CloudEvent.</b>
+     * @param source The UUri source address for the message
+     * @param attributes The UMessage attributes
+     * @param payload The UMessage payload
+     * @return returns the cloud event
+     */
+    static CloudEvent fromMessageParts(UUri source, UAttributes attributes, UPayload payload) {
+        Objects.requireNonNullElse(source, UUri.getDefaultInstance());
```

Comment:
I assume you want to assign the default value if an argument is null. Should be like:
source = Objects.requireNonNullElse(source, UUri.getDefaultInstance());

Reply:
fixed